gdpr/models.py

```diff
@@ -21,27 +79,72 @@ def create_from_purpose_slug(self, purpose_slug, source_object, issued_at=None,
                 expires_at=issued_at + purpose.expiration_timedelta,
                 tag=tag,
                 purpose_slug=purpose_slug,
-                source_object=source_object
+                source_object=source_object,
+                is_active=True
             )
             for related_object in related_objects or ():
                 legal_reason.related_objects.create(object=related_object)
             return legal_reason
         except KeyError:
             raise KeyError('Purpose with slug {} does not exits'.format(purpose_slug))
 
-    def exists_valid_consent(self, purpose_slug, source_object):
-        return LegalReason.objects.filter(
-            source_object_content_type=ContentType.objects.get_for_model(source_object.__class__),
-            source_object_id=str(source_object.pk),
-            purpose_slug=purpose_slug,
-            is_active=True,
-            expires_at__gte=timezone.now(),
-        ).exists()
+    def _refresh_for_source_by_slug(self, purpose_slug, source_object, tag=None, related_objects=None):
+        """
+        Update Legal Reason identified by purpose slug for source object
+
+        Args:
+            purpose_slug: String of Legal Reason purpose
+            source_object: Source object this Legal Reason is related to
+            tag:
+            related_objects:
+
+        Returns:
+            LegalReason: LeagalReason object
+
+        Raises:
+            MultipleObjectsReturned: If more consents with same `purpose_slug` exist
+            DoesNotExist: If consent with given source object and `purpose_slug` does not exist
+        """
+        purpose = purposes_map[purpose_slug]
+
+        try:
+            legal_reason = self.filter_source_instance(source_object).filter(purpose_slug=purpose_slug).get()
+
+            legal_reason.change_and_save(expires_at=now() + purpose.expiration_timedelta, tag=tag, is_active=True)
+
+            for related_object in related_objects or ():
+                legal_reason.related_objects.create(object=related_object)
+
+            return legal_reason
+        except LegalReason.MultipleObjectsReturned:
+            raise LegalReason.MultipleObjectsReturned('Found more Legal Reasons for object {} with purpose slug {}.'.format(
```

Comment:
to se může stát?

Reply:
myslel jsem, že proto jsi tam dával unique_together

Reply:
Tak teď už ne - to rozhodnutí, že bude jenom jeden LegalReason s daným slugem pro content_type přišlo až v pozdější fázi vývoje - takže by se dalo říct, že je to zapomenutý artefakt.